### PR TITLE
Remove peekWith/pokeWith from Unboxed module

### PR DESCRIPTION
**Pull request by rnjtranjan**

### CODE REVIEW

**Comment by rnjtranjan** [comment]
Fix:https://github.com/composewell/streamly/issues/2201

File: core/src/Streamly/Internal/Data/Unboxed.hs [hunk 1]
```diff
@@ -7,8 +7,8 @@
 
 module Streamly.Internal.Data.Unboxed
     ( Unbox(..)
-    , peekWith
-    , pokeWith
+    --, peekWith
+    --, pokeWith
```
Comment (adithyaov) [inline core/src/Streamly/Internal/Data/Unboxed.hs:11]: Remove this.

File: core/src/Streamly/Internal/Data/Unboxed.hs [hunk 2]
```diff
@@ -485,15 +485,15 @@ instance Unbox Bool where
 --------------------------------------------------------------------------------
 -- Functions
 --------------------------------------------------------------------------------
-
+{-
```
Comment (adithyaov) [inline core/src/Streamly/Internal/Data/Unboxed.hs:488]: Remove this.

File: core/src/Streamly/Internal/Data/IORef/Unboxed.hs [hunk 3]
```diff
@@ -42,11 +42,12 @@ import Data.Proxy (Proxy(..))
 import Control.Monad.IO.Class (MonadIO(..))
 import Streamly.Internal.Data.Unboxed
     ( MutableByteArray(..)
-    , Unbox
+    , Unbox (peekByteIndex)
     , sizeOf
-    , peekWith
-    , pokeWith
+    --, peekWith
+    --, pokeWith
```
Comment (adithyaov) [inline core/src/Streamly/Internal/Data/IORef/Unboxed.hs:48]: Remove this

File: core/src/Streamly/Internal/Data/IORef/Unboxed.hs [hunk 4]
```diff
@@ -42,11 +42,12 @@ import Data.Proxy (Proxy(..))
 import Control.Monad.IO.Class (MonadIO(..))
 import Streamly.Internal.Data.Unboxed
     ( MutableByteArray(..)
-    , Unbox
+    , Unbox (peekByteIndex)
```
Comment (adithyaov) [inline core/src/Streamly/Internal/Data/IORef/Unboxed.hs:45]: You can import the entire type class here. `Unbox(..)`

**Review by harendra-kumar — APPROVED** [review]

**Pull request merged**